dask_cuda/proxy_object.py

```diff
@@ -285,6 +286,7 @@ def serialize(self, serializers: Iterable[str]) -> Tuple[dict, list]:
         self.serializer = header["serializer"]
         return self.obj
 
+    @nvtx_annotate("JIT_UNSPILL", color="green", domain="dask_cuda")
```

Comment:
I think you may want to annotate `serialize` just above as well, or is there a reason I don't immediately see why you intentionally skipped that?

Reply:
I think the `JIT_SPILL_DEVICE_TO_HOST` and `JIT_SPILL_HOST_TO_DISK` annotations I added give more information than we would get by annotating `serialize`. Would also love for Mads to weigh in here.

Reply:
I don't have a strong opinion but notice, `serialize` and `deserialize` are called both when spilling and doing TCP-communication whereas `maybe_evict_from_device` and `maybe_evict_from_host` are called only when spilling.

Reply:
I agree that we don't want to annotate TCP serialization. If anyone would like to take this PR over and add the annotations in the right place, that'd be welcome!

Reply:
I think that the annotations on `maybe_evict_from_device/host` cover the movement of data from device memory to host (and then disk) [the spilling part of the equation]. I think @shwina's approach here was to handle the movement of spilled data _back_ into device memory.

I haven't managed to follow the logic of who does the unspilling yet, but presumably it happens in `ProxifyHostFile.__getitem__` ?

Reply:
> I haven't managed to follow the logic of who does the unspilling yet, but presumably it happens in ProxifyHostFile.__getitem__ ?

It is done JIT by [`ProxyObject._pxy_deserialize`](https://github.com/rapidsai/dask-cuda/blob/branch-22.12/dask_cuda/proxy_object.py#L412) when the proxy object is accessed.

Reply:
I've moved the annotation to `_pxy_deserialize` -- does this look right to you, @madsbk?

Reply:
Adding a `do-not-merge` until this conversation is resolved.